[SERV-391] Redirect unauthorized tiered access image request to degraded

markmatney opened the pull request:
Unauthorized tiered access image requests will now result in a 302 redirect to a degraded version, rather than a 401.

I anticipate that the GitHub Actions build will result in failing tests of the 401 responses for _information_ requests, since the response JSON schema has recently changed in the Cantaloupe upstream. My hope is that that change is reverted at some point, but the question of what to do in the meantime with the failing tests remains.

Commit:
[SERV-391] Redirect unauthorized tiered access image request to degraded

ksclarke commented:
Yes, I would think this is in limbo until the upstream responds or we decide to apply patches to 5.0.0 as a part of our build?

markmatney commented:
Looking at the error log, it appears that the new 403s with JSON don't have their Content-Type header set, which is causing the NoSuchElementException (TestUtils, line 54). Even if they did, though, the assertion would have failed since it's expecting non-JSON.

Commit:
Use cantaloupe-ucla:5.0.5 image for testing

This image uses a build of Cantaloupe with these bugfix patches applied:
https://github.com/UCLALibrary/docker-cantaloupe/tree/main/src/main/docker/patches

Commit:
Update vendored Cantaloupe to 5.0.5
